### PR TITLE
Limit width of main menu buttons on tablets

### LINKED ISSUE #5593: Limit width of main menu on tablets
Main menu buttons currently stretch to fill available space. On a wide screen:
tablet_buttons

@alyblenkin do you think this is best for usability? Maybe adding a max width and centering the buttons would be better?

Previously:
old landing

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #5593 

#### What has been done to verify that this works as intended?
I've tested the changes on a device with a bigger screen.

#### Why is this the best possible solution? Were any other approaches considered?
I've added `app:layout_constraintWidth_max` to every button. Instead, I could have added another layout to wrap all those buttons and only there set `app:layout_constraintWidth_max` but I didn't want to make the layout more complex.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
I reworked the entire layout used to display the main menu so please make sure there are no changes/regressions apart from limiting the width of the buttons.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/res/layout/main_menu.xml [hunk 1]
```diff
@@ -54,6 +55,7 @@
                 android:id="@+id/review_data"
                 android:layout_width="0dp"
                 android:layout_height="wrap_content"
+                app:layout_constraintWidth_max="460dp"
```
Comment (seadowg) [inline collect_app/src/main/res/layout/main_menu.xml]: An alternative to declaring the max width on each button would be to declare it for the first one and then constrain each other button (using `constraintStart_toStartOf` and `constraintEnd_toEndOf`) to that. That still means a lot of XML, but does mean you can play with that max width by changing just one attribute.

File: collect_app/src/main/res/layout/main_menu.xml [hunk 2]
```diff
@@ -1,106 +1,135 @@
...
-                android:layout_width="match_parent"
+            <org.odk.collect.android.mainmenu.StartNewFormButton
+                android:id="@+id/enter_data"
+                android:layout_width="0dp"
                 android:layout_height="wrap_content"
-                android:orientation="vertical"
-                android:paddingTop="@dimen/margin_small"
-                android:paddingBottom="@dimen/margin_standard">
+                android:layout_marginTop="@dimen/margin_small"
+                app:layout_constraintWidth_max="460dp"
```
Comment (seadowg) [inline collect_app/src/main/res/layout/main_menu.xml]: @alyblenkin how do you feel about this being `640dp` instead, so it's the same max width as the form end page? Then, we could extract this as a common dimension.

**Review by seadowg — CHANGES_REQUESTED** [review]

File: collect_app/src/main/res/layout/main_menu.xml (hunk 1 above)
Comment (grzesiek2010) [reply collect_app/src/main/res/layout/main_menu.xml]: Right, done.

File: collect_app/src/main/res/layout/main_menu.xml (hunk 2 above)
Comment (alyblenkin) [reply collect_app/src/main/res/layout/main_menu.xml]: So sorry I missed this! Yes, this sounds good!

Comment (grzesiek2010) [reply collect_app/src/main/res/layout/main_menu.xml]: Done.

File: collect_app/src/main/res/layout/main_menu.xml [hunk 3]
```diff
@@ -1,106 +1,130 @@
...
-                android:layout_width="match_parent"
+            <org.odk.collect.android.mainmenu.StartNewFormButton
+                android:id="@+id/enter_data"
+                android:layout_width="0dp"
                 android:layout_height="wrap_content"
-                android:orientation="vertical"
-                android:paddingTop="@dimen/margin_small"
-                android:paddingBottom="@dimen/margin_standard">
+                android:layout_marginTop="@dimen/margin_small"
+                app:layout_constraintWidth_max="640dp"
```
Comment (seadowg) [inline collect_app/src/main/res/layout/main_menu.xml]: Just before we move on, could this get extracted to `dimens.xml`? We also use it on the form end view as the max width there as well.

**Review by seadowg — CHANGES_REQUESTED** [review]

File: collect_app/src/main/res/layout/main_menu.xml (hunk 3 above)
Comment (grzesiek2010) [reply collect_app/src/main/res/layout/main_menu.xml]: What name do you think would be best for it?

Comment (seadowg) [reply collect_app/src/main/res/layout/main_menu.xml]: `max_content_width` maybe?

**Review by seadowg — APPROVED** [review]

**Comment by dbemke** [comment]
**Tested with Success!**

Verified on device with Android 8.1, 10, emulated 7.0 and emulated tablet with Android 8.1

**Verified cases:**
- main menu layout according to #5593
- landscape and portrait view
- changing font size
- multiclicking
- dark and light mode
- scrolling the menu
- rotating

**Comment by srujner** [comment]
**Tested with Success!**

Verified on device with Android 13, emulated tablet with Android 12 on Pixel C device

**Pull request merged**